.rubocop.yml

```diff
@@ -1,22 +1,34 @@
 # use the shared YaST defaults
 inherit_from:
-  /usr/share/YaST2/data/devtools/data/rubocop-0.71.0_yast_style.yml
+  /usr/share/YaST2/data/devtools/data/rubocop-1.24.1_yast_style.yml
 
 # Don't enforce any particular name for block params
-SingleLineBlockParams:
+Style/SingleLineBlockParams:
   Enabled: false
 
 # The general YaST agreement is 100 characters, so it fits into standard
 # resolutions and Github's pull request view. But let's open the threshold a
 # little bit
-Metrics/LineLength:
+Layout/LineLength:
   Max: 105
 
 Metrics/BlockLength:
   # rspec is known as DSL with big blocks
   Exclude:
     - test/**/*
 
+# TODO: fix, but for now too much work
```

Comment:
Why exactly these three need to be disabled now? Is the new default limit more strict? Were the rules refined and now we have methods that rank higher than they used to rank? If so, how many?

Reply:
well, it counts more precisely.
mainly https://github.com/rubocop/rubocop/blob/master/CHANGELOG.md#changes-36 do better counting of conditions. and similar at https://github.com/rubocop/rubocop/blob/master/CHANGELOG.md#changes-39

So I think it is fixable, I just do not want to spend too much time in it. So if it is blocking I can check it and do refactoring.

Reply:
I see. Then I guess it makes sense to raise the limit. I just feel we raised them too much. The new default values are 7 and 17 and we raised ours to 12 and 32.

Let's say we keep the `rubocop:disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity` directives that you deleted in some parts (those are methods somebody already considered not worth refactoring). Would we still need to go as far as 12 and 32?

Reply:
I found that now biggest offender is https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2storage/proposal_settings.rb#L378 as it counts each ||= as condition. I can try to refactor it, if you want. Btw also as it uses booleans and `||= true` looks quite bad from my POV.